### PR TITLE
Remove zero-value assets from accounts after operating upon.

### PR DESCRIPTION
**Pull request by Revertron**
### Description of the Change

Remove assets that have zero value after some operations on them.
For example, if user has moved all assets elsewhere.

### Benefits

It will give storage and performance improvement.

### Possible Drawbacks 

Checks are performed after any operation. Could they slowdown or not?

### CODE REVIEW

File: iroha/src/wsv.rs [hunk 1]
```diff
@@ -241,7 +241,18 @@ impl WorldStateView {
...
+            let mut keys_to_remove = Vec::new();
+            account.assets.iter().for_each(|(id, asset)| {
+                if asset.value.is_zero() {
+                    keys_to_remove.push(id.clone());
+                }
+            });
+            for id in keys_to_remove {
+                drop(account.assets.remove(&id));
+            }
+            Ok(())
```
Comment (i1i1) [inline iroha/src/wsv.rs:255]: This seems like O(n) in asymptotics. I guess there should be constant ways of prunning assets (may be inside of instructions themselves?)

Comment (i1i1) [reply iroha/src/wsv.rs:255]: This operation should be O(1). May be we can even add it in doc-comment

Comment (Revertron) [reply iroha/src/wsv.rs:255]: Hmm, we could check only that one asset, that we (presumably) changed, you're right!

Comment (Revertron) [reply iroha/src/wsv.rs]: Fixed.

**Review by i1i1 — APPROVED** [review]

File: iroha_data_model/src/lib.rs [hunk 2]
```diff
@@ -1203,6 +1203,14 @@ pub mod asset {
...
             }
         }
+        /// Returns true if this value is zero, false if it contains [`Metadata`] or positive value
+        pub const fn is_zero_value(&self) -> bool {
+            match *self {
+                AssetValue::Quantity(q) => q == 0_u32,
+                AssetValue::BigQuantity(q) => q == 0_u128,
+                AssetValue::Store(_) => false,
+            }
+        }
```
Comment (Revertron) [inline iroha_data_model/src/lib.rs:1213]: Changed this function to return `false` it the value is metadata. And renamed it accordingly.
Now tests are happy.

**Review by i1i1 — APPROVED** [review]

**Pull request merged**